internal/powerbi/data_source_capacity.go

```diff
@@ -0,0 +1,63 @@
+package powerbi
+
+import (
+	"fmt"
+
+	"github.com/codecutout/terraform-provider-powerbi/internal/powerbiapi"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+// DataSourceCapacity represents a Power BI capacity
+func DataSourceCapacity() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceCapacityRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the capacity.",
+			},
+			"sku": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "SKU of the capacity",
+			},
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: "Region of the capacity.",
+			},
+		},
+	}
+}
+
+func dataSourceCapacityRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*powerbiapi.Client)
+	name := d.Get("name").(string)
+	capacityList, err := client.GetCapacities()
+	if err != nil {
+		return err
+	}
+	var capacityObjFound bool
+	var capacityObj powerbiapi.GetCapacitiesResponseItem
+	if len(capacityList.Value) >= 1 {
+		for _, capacityObj = range capacityList.Value {
+			if capacityObj.DisplayName == name {
+				capacityObjFound = true
```

Comment:
You are missing a `break` here. Otherwise, if you have more than one capacity in your return list, you will always ever get the last one returned, no matter which name matched.